release-notes/6.0/preview/api-diff/preview5/.Net/6.0-preview5_System.Security.Cryptography.md

```diff
@@ -0,0 +1,420 @@
+# System.Security.Cryptography
+
+``` diff
+ namespace System.Security.Cryptography {
+     public sealed class AesCcm : IDisposable {
++        public static bool IsSupported { get; }
+     }
++    public sealed class AesCng : Aes {
++        public AesCng();
++        public AesCng(string keyName);
++        public AesCng(string keyName, CngProvider provider);
++        public AesCng(string keyName, CngProvider provider, CngKeyOpenOptions openOptions);
++        public override byte[] Key { get; set; }
++        public override int KeySize { get; set; }
++        public override ICryptoTransform CreateDecryptor();
++        public override ICryptoTransform CreateDecryptor(byte[] rgbKey, byte[] rgbIV);
++        public override ICryptoTransform CreateEncryptor();
++        public override ICryptoTransform CreateEncryptor(byte[] rgbKey, byte[] rgbIV);
++        protected override void Dispose(bool disposing);
++        public override void GenerateIV();
++        public override void GenerateKey();
++    }
+     public sealed class AesGcm : IDisposable {
++        public static bool IsSupported { get; }
+     }
++    public sealed class ChaCha20Poly1305 : IDisposable {
++        public ChaCha20Poly1305(byte[] key);
++        public ChaCha20Poly1305(ReadOnlySpan<byte> key);
++        public static bool IsSupported { get; }
++        public void Decrypt(byte[] nonce, byte[] ciphertext, byte[] tag, byte[] plaintext, byte[] associatedData = null);
++        public void Decrypt(ReadOnlySpan<byte> nonce, ReadOnlySpan<byte> ciphertext, ReadOnlySpan<byte> tag, Span<byte> plaintext, ReadOnlySpan<byte> associatedData = default(ReadOnlySpan<byte>));
++        public void Dispose();
++        public void Encrypt(byte[] nonce, byte[] plaintext, byte[] ciphertext, byte[] tag, byte[] associatedData = null);
++        public void Encrypt(ReadOnlySpan<byte> nonce, ReadOnlySpan<byte> plaintext, Span<byte> ciphertext, Span<byte> tag, ReadOnlySpan<byte> associatedData = default(ReadOnlySpan<byte>));
++    }
++    public sealed class CngAlgorithm : IEquatable<CngAlgorithm> {
++        public CngAlgorithm(string algorithm);
```

Comment:
Same question here... hasn't a lot of this stuff previously existed?

Reply:
It seems like we added openssl and cng dlls to the ref pack cc @ericstj @ViktorHofer